main.go

```diff
@@ -33,19 +48,56 @@ func main() {
 		_ = resp.Body.Close()
 	}()
 
+	// Check that the domain specified by the user exists in GoDaddy
+	domainExists := 0
 	for _, zone := range zones {
-		if zone.Status == "ACTIVE" {
-			log.Println(zone.Domain)
+		if strings.EqualFold(zone.Status, "active") && strings.EqualFold(zone.Domain, *userDomainName) {
+			domainExists = 1
+			log.Println("Domain", zone.Domain, "matches user domain", *userDomainName)
 		}
 	}
 
-	records, _, err := apiClient.V1Api.RecordGet(ctx, "lwts.org", "", "", nil)
+	if domainExists != 1 {
+		log.Println("Domain", *userDomainName, "does not match any domains returned from the API")
+		os.Exit(1)
+	}
+
+	var recordData map[string]string
+	recordData = make(map[string]string)
+
+	records, _, err := apiClient.V1Api.RecordGet(ctx, *userDomainName, "", "", nil)
 	for _, record := range records {
-		log.Println(record.Name, "=>", record.Data)
+		if record.Name == "@" {
+			if record.Type_ == "A" {
+				log.Println("Changing", record.Type_, record.Data, "to", *userIPAddress)
+				// Fill in the recordData map so it can be used outside this loop
+				recordData["type"] = "A"
+				recordData["name"] = "@"
+
+			}
+		}
+		//log.Println(record.Name, "=>", record.Data)
 	}
 
-	domainInfo, _, err := apiClient.V1Api.Get(ctx, "lwts.org", nil)
-	log.Println(domainInfo)
+	if *applyBool {
```

Comment:
Always keep good code to the left. The "normal" path should be left. Outlier paths should branch out. In this case, it appears that the code in this branch is the good path, so re-write this if statement like so:
```golang
if !*applyBool {
	log.Println("The --apply bool is required to apply the changes")
	os.Exit(1)
}
```

Then the rest of the code can be to the left, without being in a branched `if`  stanza. Bonus: you remove the `else`.